activities/Write.activity/js/activity.js

```diff
@@ -43,19 +43,36 @@ define([
 		// Journal Handling (Load)
 		env.getEnvironment(function(err, environment) {
             
-        	currentenv = environment;
-            
+			currentenv = environment;
+			// Language Settings
+			var defaultLanguage = (typeof chrome != 'undefined' && chrome.app && chrome.app.runtime) ? chrome.i18n.getUILanguage() : navigator.language;
+            var language = environment.user ? environment.user.language : defaultLanguage;
+			webL10n.language.code = language;
+			
 			if (!environment.objectId && !environment.sharedId) {
 				// New instance
-				var greet = 'Welcome ! ';
-				var name = environment.user.name + ' ';
-				editor.setContents([
-					{ insert:  greet, attributes: { size: "40px" , color : environment.user.colorvalue.stroke , bold: true }},
-					{ insert: name, },
-					{ insert: '\n' }
-				]);
-				var index = greet.length + name.length ;
-				editor.setSelection(index,0);				
+				console.log(window.initialImageDataUrl);
+				// Intentionally added setTimeout to allow locale.ini file to be loaded
+				setTimeout(function(){
+					editor.setContents([
```

Comment:
Unfortunately, the way of building a sentence is not the same in all languages. So you can't simply use a list of inserts and be sure that it could work in any language to rebuild the sentence. You can only insert sentence by sentence, not word by word.
So I suggest two things:
- For the first sentence, use a webL10n variable into the .INI string for the username, something like `Welcome {{username}}!`. You should pass the username value to the webL10n get call.
- For other sentences, you should use a substitution into the .INI string, something like `Write is an editor where you could [b]type[/b] and [i]format[/i] your [u]text[/u] and ideas.` Then call the webL10n get function and split the result to add specific QuillJS attributes on each part resulting of the sentence.